Dockerfile

```diff
@@ -59,24 +36,24 @@ RUN go install github.com/gogo/protobuf/protoc-gen-gogofaster
 RUN go install github.com/gogo/protobuf/protoc-gen-gogoslick
 RUN go install github.com/gogo/protobuf/protoc-gen-gogotypes
 RUN go install github.com/gogo/protobuf/protoc-gen-gostring
-RUN go get github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema
 RUN go install github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema
-RUN go install github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+RUN go install github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway
 RUN go install github.com/envoyproxy/protoc-gen-validate
 RUN go install github.com/mwitkow/go-proto-validators/protoc-gen-govalidators
 RUN go install github.com/pseudomuto/protoc-gen-doc/cmd/...
 RUN go install github.com/infobloxopen/protoc-gen-preprocess
-RUN go install  \
-      -ldflags "-X github.com/infobloxopen/protoc-gen-gorm/plugin.ProtocGenGormVersion=$PGG_VERSION -X github.com/infobloxopen/protoc-gen-gorm/plugin.AtlasAppToolkitVersion=$AAT_VERSION" \
-      github.com/infobloxopen/protoc-gen-gorm
-# Download all dependencies of protoc-gen-atlas-query-validate
-RUN cd ${GOPATH}/src/github.com/infobloxopen/protoc-gen-atlas-query-validate && dep ensure -vendor-only
 RUN go install github.com/infobloxopen/protoc-gen-atlas-query-validate
-
-# Download all dependencies of protoc-gen-atlas-validate
-RUN cd ${GOPATH}/src/github.com/infobloxopen/protoc-gen-atlas-validate && dep ensure -vendor-only
 RUN go install github.com/infobloxopen/protoc-gen-atlas-validate
 
+# TODO: this should be installed the same way once it is compatible with updated protobuf
+RUN GO111MODULE=on go install  \
+      -ldflags "-X github.com/infobloxopen/protoc-gen-gorm/plugin.ProtocGenGormVersion=$PGG_VERSION -X github.com/infobloxopen/protoc-gen-gorm/plugin.AtlasAppToolkitVersion=$AAT_VERSION" \
+      github.com/infobloxopen/protoc-gen-gorm@$PGG_VERSION
+
+# Download any projects that have proto-only packages, since go mod ignores those
+RUN cd ${GOPATH}/src/github.com/infobloxopen && rm -rf protoc-gen-gorm && git clone https://github.com/infobloxopen/protoc-gen-gorm && cd protoc-gen-gorm && git checkout $PGG_VERSION
+RUN cd ${GOPATH}/src/github.com && mkdir googleapis && cd googleapis && git clone https://github.com/googleapis/googleapis
```

Comment:
Copying these files into third party is the preferred way to manage them. See this example https://github.com/grpc-ecosystem/grpc-gateway/tree/v2.2.0/third_party/
This reduces risk of downloading things from CI as well.

Reply:
Interesting. It is worth noting that the only files copied to the final output are proto files and the `protoc-gen-*` executables, so that should only affect the intermediate step.